source/core/Events.cpp

```diff
@@ -548,6 +549,30 @@ static inline EventSource GetEventSourceForEventType(EventType eType) {
     return eSource;
 }
 
+void GetVIName(VirtualInstrument *vi, StringRef viName) {
+    PercentEncodedSubString encodedStr(vi->VIName(), true, false);
+    SubString encodedSubstr = encodedStr.GetSubString();
+    viName->AppendSubString(&encodedSubstr);
+}
+
+void RegisterForControlEvent(EventInfo *eventInfo, EventSpec eventSpec, StringRef viName, EventQueueID qID, UInt32 controlID, RefNum reference, bool registerControlEvent) {
```

Comment:
Similar helper as above. This will end up calling jsRegisterForcontrolEvent for ControlRefNums, but not JavaScriptRefNums. We may be able to get away without the RegisterForJSEvent instruction if we just always attempt to register here. It should still only do this once like for ControlRefNums.